index/secondary-index/src/test/scala/org/apache/carbondata/spark/testsuite/secondaryindex/TestSIWithSecondryIndex.scala

```diff
@@ -86,6 +86,64 @@ class TestSIWithSecondryIndex extends QueryTest with BeforeAndAfterAll {
       .contains("Alter table drop column operation failed:"))
   }
 
+  test("test create secondary index global sort after insert") {
+    sql("drop table if exists table1")
+    sql("create table table1 (name string, id string, country string) stored as carbondata")
+    sql("insert into table1 select 'xx', '2', 'china' union all select 'xx', '1', 'india'")
+    sql("create index table1_index on table table1(id, country) as 'carbondata' properties" +
+        "('sort_scope'='global_sort', 'Global_sort_partitions'='3')")
+    checkAnswerWithoutSort(sql("select id, country from table1_index"),
+      Seq(Row("1", "india"), Row("2", "china")))
+    // check for valid sort_scope
+    checkExistence(sql("describe formatted table1_index"), true, "Sort Scope global_sort")
+    // check the invalid sort scope
+    assert(intercept[MalformedCarbonCommandException](sql(
+      "create index index_2 on table table1(id, country) as 'carbondata' properties" +
+      "('sort_scope'='tim_sort', 'Global_sort_partitions'='3')"))
+      .getMessage
+      .contains("Invalid SORT_SCOPE tim_sort"))
+    // check for invalid global_sort_partitions
+    assert(intercept[MalformedCarbonCommandException](sql(
+      "create index index_2 on table table1(id, country) as 'carbondata' properties" +
+      "('sort_scope'='global_sort', 'Global_sort_partitions'='-1')"))
+      .getMessage
+      .contains("Table property global_sort_partitions : -1 is invalid"))
+    sql("drop index table1_index on table1")
```

Comment:
can just do drop table, it will drop index too, no need to separately run drop index and suggest to give a better tableName and index name and please check other test for same input.

Reply:
a) I know, but calling drop index will not add extra overhead.
b) For test cases these names are enough ! This is not an example file.

Reply:
a) not talking about overhead, why to call the command when that will be handled by drop table, why to take effort to call another command, please remove it and same for other test case.
b) even though its not an example file, we  should always give proper and meaningful names. Just because user uses carbon and see code, we cant give non meaningful names right...!!!

Reply:
>  we cant give non meaningful names right...!!!
table1 is a meaningful name to represent it as a table, it is like john wick calling his dog as a dog. On a lighter note, Stop focusing on unimportant things. As an experienced developer, I do know when a code is not readable.

Reply:
its not about the experience, but i always prefer the code to be very clean and meaningful and reader or developer should be happy reading about it and clean and meaningful names are very important aspect in any code...!

Reply:
I can see that from your PR #3608 , you have used t1 as table name and i1 as index name in DropTableTest. is that clean and meaningful name ? I don't want to argue further. 
Table1 is still a table name I have not named it as car or bike.